gradoop-temporal/src/main/java/org/gradoop/temporal/model/impl/operators/keyedgrouping/keys/TimeIntervalKeyFunction.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © 2014 - 2019 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.temporal.model.impl.operators.keyedgrouping.keys;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.typeutils.TupleTypeInfo;
+import org.gradoop.flink.model.api.functions.KeyFunction;
+import org.gradoop.temporal.model.api.functions.TimeDimension;
+import org.gradoop.temporal.model.impl.pojo.TemporalElement;
+
+import java.util.Objects;
+
+/**
+ * A key function extracting a {@link TimeDimension} from a {@link TemporalElement}.
```

Comment:
It extracts a time interval specified by a `TimeDimension`, not a `TimeDimension` itself.